### PR TITLE
Separate debugging workspace service and EnC service

### PR DESCRIPTION
**Pull request by tmat**
### Customer scenario

Refactoring with no customer visible changes.
Part of EnC manager rewrite: https://github.com/dotnet/roslyn/issues/18350

Separates debugging workspace service and EnC service so that EnC service can add dependency on MEF components only exported in Visual Studio and not available in other hosts.

### Bugs this fixes

N/A

### Workarounds, if any

N/A

### Risk

Small.

### Performance impact

No impact

### Is this a regression from a previous update?

No

### Root cause analysis

N/A

### How was the bug found?

N/A

### CODE REVIEW

**Comment by tmat** [comment]
@jasonmalinowski @ivanbasov Please review.

File: src/EditorFeatures/Test2/Rename/InlineRenameTests.vb [hunk 1]
```diff
@@ -1389,8 +1389,8 @@ class C
                 Await VerifyTagsAreCorrect(workspace, "BarGoo")
 
                 ' Simulate ending break mode in the debugger (by stepping or continuing)
-                Dim editAndContinueWorkspaceService = workspace.Services.GetService(Of IEditAndContinueWorkspaceService)
-                editAndContinueWorkspaceService.OnBeforeDebuggingStateChanged(DebuggingState.Break, DebuggingState.Run)
+                Dim debuggingService = workspace.Services.GetService(Of IDebuggingWorkspaceService)
```
Comment (ivanbasov) [inline src/EditorFeatures/Test2/Rename/InlineRenameTests.vb]: >debuggingService [](start = 20, length = 16)

nit: debuggingWorkspaceService for consistency

**Comment by tmat** [comment]
@Pilchie for approval.

**Comment by Pilchie** [comment]
I'll need an ask mode template please.

**Comment by tmat** [comment]
@Pilchie OK. Although I don't know what to fill it with other than N/A.

**Comment by tmat** [comment]
test failure: https://github.com/dotnet/roslyn/issues/23657

**Comment by tmat** [comment]
test windows_release_unit32_prtest please

**Pull request merged**